packages/core/src/errors/InruptError.ts

```diff
@@ -21,60 +21,107 @@
+ *
+ * Error Identifiers
+ *   Our error class supports the notion of globally unique, dereferenceable
+ *   error identifiers in the form of IRIs.
+ *
+ * Parameterized error messages
+ *   This class supports error messages with positional placeholders that can be
+ *   replaced by developer-provided parameter values at runtime.
+ *
+ * Wrapped Errors (Coming soon!)
+ *   Similar to how Java supports the notion of wrapped exceptions, our error
+ *   class implementation supports the ability to wrap [[Error]]s.
+ *
+ * HTTP errors:
+ *   This class provides a very convenient means to provide a HTTP response,
+ *   the details from which can be appended to error message strings, and the
+ *   response itself retrieved directly by consumers of instances of this class.
  * @packageDocumentation
  */
 
-/**
- * In a subsequent PR, just re-export needed implementations from generated
- * artifacts directly, e.g. like @inrupt/solid-common-vocab, e.g. in the
- * `index.hbs` template of RDF/JS artifacts include this:
- // Re-export some of the basic RDF/JS implementations that we use ourselves -
- // allows users of our generated artifacts to easily import these
- // implementations if they want (e.g. in their tests, or to instantiate
- // hard-coded IRIs for convenience (as they shouldn't really need to do that
- // much, instead using our generated constants in most cases)).
- export { NamedNode } from "rdf-js";
- *  Then we can either import that implementation in this code, or just import
- *  the types when needed here, e.g.:
- import type { NamedNode } from "rdf-js";
- */
-import { NamedNode } from "n3";
-
-import { VocabTerm } from "@inrupt/solid-common-vocab";
+import { VocabTerm, NamedNode } from "@inrupt/solid-common-vocab";
 
 export default class InruptError extends Error {
-  public readonly statusCode?: number;
-
-  public readonly statusText?: string;
+  // For the common case of HTTP errors, we can store the HTTP response to allow
+  // consumers of this error instance to access it directly.
+  // NOTE: we specifically stipulate that we expect the HTTP response to be an
+  // error response!
+  private httpErrorResponse?: Response & { ok: false };
 
   constructor(
     messageOrIri: string | NamedNode | VocabTerm,
     messageParams?: string[],
-    httpResponse?: Response & { ok: false },
-    appendHttpDetails = true,
     appendErrorIri = true
   ) {
     super(
-      InruptError.appendHttpResponseDetails(
-        typeof messageOrIri === "string"
-          ? InruptError.substituteParams(messageOrIri, messageParams)
-          : InruptError.appendErrorIri(
-              InruptError.lookupErrorIri(messageOrIri, messageParams),
-              messageOrIri as NamedNode,
-              appendErrorIri
-            ),
-        httpResponse,
-        appendHttpDetails
-      )
+      typeof messageOrIri === "string"
+        ? InruptError.substituteParams(messageOrIri, messageParams)
```

Comment:
I'm not sure where `substituteParams` comes from, but I don't see any implementation... Should we consider using a proper templating library ?

Reply:
The function `substituteParams()` is just a static function at the bottom of the file. Yeah, we could use a proper templating library if it brought something we needed, but at the moment this function is just 17 lines of code (which includes some nice error checking), so for now I'd certainly rather those 17 lines of code over bringing in another dependency.